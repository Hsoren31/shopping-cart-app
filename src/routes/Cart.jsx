import { Link } from "react-router-dom";
import styles from "../styles/Cart.module.css";
import { useOutletContext } from "react-router-dom";
import CartItem from "../components/CartItem.jsx";

function Cart() {
  const { cartContents } = useOutletContext();

  if (cartContents.quantity > 0) {
    return (
      <div className={styles.cartItems}>
        {cartContents.items.map((product) => (
          <CartItem product={product} key={product.id} />
        ))}
      </div>
    );
  }
  return (
    <div className={styles.empty}>
      <p className={styles.emptyText}>Looks like your cart is empty!</p>
      <br></br>
      <Link className={styles.shopLink} to="/shop">
        Start Shopping
      </Link>
    </div>
  );
}

export default Cart;
