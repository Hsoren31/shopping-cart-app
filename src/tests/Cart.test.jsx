import { screen, render, getByTestId } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import { RenderRouteWithOutletContext } from "./RenderRouteWithOutletContext";
import Cart from "../routes/Cart";

const mockEmptyCart = {
  cartContents: {
    quantity: 0,
    items: [],
    price: 0,
  },
};

const mockNotEmptyCart = {
  cartContents: {
    items: [
      {
        id: 1,
        title: "foo",
      },
    ],
    quantity: 1,
    price: 10,
  },
  formatPrice: vi.fn(),
};

const mockMultipleItemsCart = {
  cartContents: {
    items: [
      {
        id: 1,
        title: "foo",
      },
      {
        id: 1,
        title: "bar",
      },
      {
        id: 1,
        title: "foobar",
      },
    ],
    quantity: 3,
    price: 30,
  },
  formatPrice: vi.fn(),
};

describe("Cart Route", () => {
  it("Should show message and go shopping btn with an empty cart", () => {
    render(
      <RenderRouteWithOutletContext context={mockEmptyCart}>
        <Cart />
      </RenderRouteWithOutletContext>
    );

    expect(screen.getByRole("paragraph").textContent).toMatch(
      /looks like your cart is empty/i
    );

    expect(screen.getByText(/start shopping/i)).toBeInTheDocument();
  });

  it("Should not show message or go shopping btn when cart is not empty", () => {
    render(
      <RenderRouteWithOutletContext context={mockNotEmptyCart}>
        <Cart />
      </RenderRouteWithOutletContext>
    );

    expect(
      screen.queryByText(/looks like your cart is empty/)
    ).not.toBeInTheDocument();

    expect(screen.queryByText(/start shopping/i)).not.toBeInTheDocument();
  });

  // it("Should render one product correctly", () => {
  //   render(
  //     <RenderRouteWithOutletContext context={mockNotEmptyCart}>
  //       <Cart />
  //     </RenderRouteWithOutletContext>
  //   );

  //   const listItems = screen.getAllByRole("listitem");
  //   expect(listItems.length).toEqual(1);
  // });

  // it("Should render multiple products correctly", () => {
  //   render(
  //     <RenderRouteWithOutletContext context={mockMultipleItemsCart}>
  //       <Cart />
  //     </RenderRouteWithOutletContext>
  //   );

  //   const listItems = screen.getAllByRole("listitem");
  //   expect(listItems.length).toEqual(3);
  // });
});
