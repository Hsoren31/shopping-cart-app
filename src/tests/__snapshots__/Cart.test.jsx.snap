// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cart Route > Should render multiple products correctly 1`] = `
<div
  class="_cart_c7a017"
  data-testid="cartContainer"
>
  <ul
    class="_cartItems_c7a017"
  >
    <li
      class="_item_c7a017"
    >
      <img
        alt="foo"
        class="_image_c7a017"
      />
      <h3
        class="_title_c7a017"
      >
        foo
      </h3>
      <div
        class="_quantity_c7a017"
      >
        <button>
          -
        </button>
        <input
          min="1"
          type="number"
          value=""
        />
        <button>
          +
        </button>
      </div>
      <p>
        $undefined
      </p>
      <button>
        Remove
      </button>
    </li>
    <li
      class="_item_c7a017"
    >
      <img
        alt="bar"
        class="_image_c7a017"
      />
      <h3
        class="_title_c7a017"
      >
        bar
      </h3>
      <div
        class="_quantity_c7a017"
      >
        <button>
          -
        </button>
        <input
          min="1"
          type="number"
          value=""
        />
        <button>
          +
        </button>
      </div>
      <p>
        $undefined
      </p>
      <button>
        Remove
      </button>
    </li>
    <li
      class="_item_c7a017"
    >
      <img
        alt="foobar"
        class="_image_c7a017"
      />
      <h3
        class="_title_c7a017"
      >
        foobar
      </h3>
      <div
        class="_quantity_c7a017"
      >
        <button>
          -
        </button>
        <input
          min="1"
          type="number"
          value=""
        />
        <button>
          +
        </button>
      </div>
      <p>
        $undefined
      </p>
      <button>
        Remove
      </button>
    </li>
  </ul>
  <div
    class="_orderSummary_c7a017"
  >
    <h2>
      Order Summary
    </h2>
    <div
      class="_row_c7a017"
    >
      <p>
        3 items
      </p>
      <p>
        $undefined
      </p>
    </div>
    <div
      class="_row_c7a017"
    >
      <p>
        Standard Shipping
      </p>
      <p>
        $undefined
      </p>
    </div>
    <div
      class="_row_c7a017"
    >
      <p>
        Subtotal
      </p>
      <p>
        $undefined
      </p>
    </div>
    <button
      class="_checkout_c7a017"
    >
      Checkout
    </button>
  </div>
</div>
`;

exports[`Cart Route > Should render one product correctly 1`] = `
<div
  class="_cart_c7a017"
  data-testid="cartContainer"
>
  <ul
    class="_cartItems_c7a017"
  >
    <li
      class="_item_c7a017"
    >
      <img
        alt="foo"
        class="_image_c7a017"
      />
      <h3
        class="_title_c7a017"
      >
        foo
      </h3>
      <div
        class="_quantity_c7a017"
      >
        <button>
          -
        </button>
        <input
          min="1"
          type="number"
          value=""
        />
        <button>
          +
        </button>
      </div>
      <p>
        $undefined
      </p>
      <button>
        Remove
      </button>
    </li>
  </ul>
  <div
    class="_orderSummary_c7a017"
  >
    <h2>
      Order Summary
    </h2>
    <div
      class="_row_c7a017"
    >
      <p>
        1 items
      </p>
      <p>
        $undefined
      </p>
    </div>
    <div
      class="_row_c7a017"
    >
      <p>
        Standard Shipping
      </p>
      <p>
        $undefined
      </p>
    </div>
    <div
      class="_row_c7a017"
    >
      <p>
        Subtotal
      </p>
      <p>
        $undefined
      </p>
    </div>
    <button
      class="_checkout_c7a017"
    >
      Checkout
    </button>
  </div>
</div>
`;
